src/govuk/helpers/links.test.js

```diff
@@ -121,7 +121,7 @@ describe('@mixin govuk-link-hover-decoration', () => {
   })
 
   describe('when $govuk-new-link-styles are enabled', () => {
-    it('sets text-decoration-thickness on hover', async () => {
+    it('sets a hover state', async () => {
```

Comment:
The test has to change because it no longer passes:

```
  ● @mixin govuk-link-hover-decoration › when $govuk-new-link-styles are enabled › sets text-decoration-thickness on hover

    expect(received).toContain(expected) // indexOf

    Expected substring: ".foo:hover { text-decoration-thickness: 10px; }"
    Received string:    ".foo:hover { text-decoration-thickness: 10px; text-decoration-skip-ink: none; text-decoration-skip: none; }
```

We could add tests for each of the properties, but I don't think we want to test the CSS itself, just the logic that surrounds it. In this case, we want to know how the mixin behaves differently depending on the various link settings:

- by default there's no hover state
- when `$govuk-new-link-styles` are enabled, there is a hover state (but we don't care exactly what that hover state looks like – that's what manual testing is for)
- except when `$govuk-link-hover-underline-thickness` is falsey, in which case the hover state is removed

I think the revised tests cover this. What do you think?

Reply:
Thanks @36degrees that covers it 👍